torch/functional.py

```diff
@@ -1156,9 +1156,10 @@ def tensordot(a, b, dims=2, out: Optional[torch.Tensor] = None):  # noqa: F811
         dims_b = list(range(dims))
 
     if out is None:
-        return _VF.tensordot(a, b, dims_a, dims_b)  # type: ignore[attr-defined]
+        return torch._C._linalg.linalg_tensordot(a, b, dims_a, dims_b)  # type: ignore[attr-defined]
```

Comment:
I'm a little confused by this part still. This functional has a bunch of logic before calling `torch._C._linalg.linalg_tensordot()`, but when `torch.linalg.tensordot()` is called how does it implement all the above logic?

Reply:
`torch.linalg.tensordot` is the very same function as `torch.functional.tensordot`:
```
>>> import torch
>>> torch.linalg.tensordot is torch.functional.tensordot
True
```